### ISSUE TITLE
[Bug]: Update to Nextcloud 25.0.3 doesn't migrate theming images

### ISSUE THREAD

**Issue by DATA-HORIZON** [nextcloud/server#36206]
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on Github _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] Nextcloud Server **is** running on 64bit capable CPU, PHP and OS.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

After updating several Nextcloud instances from Nextcloud 24.0.8 to  25.0.2 and  24.0.9 to  25.0.3 we are facing always the same problem: all images under theming are removed from the file system (but not from the database). As a result the design is broken and 404's are thrown for the images lost.

### Steps to reproduce

1. add images to the theming app in your Nextcloud 24.0.9 ("Logo" / "Background and login image" / "Header logo" / "Favicon")
2. update your Nextcloud to version 25.0.3
3. enjoy the broken design

### Expected behavior

Images of the theming app should not get "lost" when upgrading to Nextcloud 25.0.3.

After investigating the issue we could determine the following work-around:

- the problem is the new directory structure of the theming app:
-  from a backup before the upgrade copy the images of the folder...
```"data/appdata_**************/theming/*"```
- ...to the new subdirectory "global/images" that is created on the update process ```"data/appdata_**************/theming/global/images"```
-  from a backup before the upgrade copy the icon/favicon directory of the folder...
```"data/appdata_**************/theming/[digits]"```
-  ...to the new subdirectory "global" that is created on the update process
```"data/appdata_**************/theming/global/[digits]"```

### Installation method

Community Manual installation with Archive

### Operating system

Debian/Ubuntu

### PHP engine version

PHP 8.0

### Web server

Apache (supported)

### Database engine version

MariaDB

### Is this bug present after an update or on a fresh install?

Updated to a major version (ex. 22.2.3 to 23.0.1)

### Are you using the Nextcloud Server Encryption module?

Encryption is Disabled

### What user-backends are you using?

- [X] Default user-backend _(database)_
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

### List of activated Apps

```shell
Enabled:
  - activity: 2.17.0
  - audioplayer: 3.3.1
  - circles: 25.0.0
  - cloud_federation_api: 1.8.0
  - contactsinteraction: 1.6.0
  - dashboard: 7.5.0
  - dav: 1.24.0
  - federatedfilesharing: 1.15.0
  - files: 1.20.1
  - files_pdfviewer: 2.6.0
  - files_rightclick: 1.4.0
  - files_sharing: 1.17.0
  - files_trashbin: 1.15.0
  - files_versions: 1.18.0
  - firstrunwizard: 2.14.0
  - logreader: 2.10.0
  - lookup_server_connector: 1.13.0
  - nextcloud_announcements: 1.14.0
  - notes: 4.6.0
  - notifications: 2.13.1
  - oauth2: 1.13.0
  - onlyoffice: 7.6.8
  - password_policy: 1.15.0
  - photos: 2.0.1
  - previewgenerator: 5.1.1
  - provisioning_api: 1.15.0
  - quota_warning: 1.15.0
  - related_resources: 1.0.4
  - serverinfo: 1.15.0
  - settings: 1.7.0
  - sharebymail: 1.15.0
  - text: 3.6.0
  - theming: 2.0.1
  - twofactor_backupcodes: 1.14.0
  - user_status: 1.5.0
  - viewer: 1.9.0
  - weather_status: 1.5.0
  - workflowengine: 2.7.0
Disabled:
  - admin_audit
  - bruteforcesettings: 2.4.0
  - camerarawpreviews: 0.8.0
  - comments: 1.8.0
  - encryption
  - federation: 1.8.0
  - files_external: 1.9.0
  - privacy: 1.0.0
  - recommendations: 0.6.0
  - support: 1.1.0
  - survey_client: 1.6.0
  - suspicious_login
  - systemtags: 1.8.0
  - twofactor_totp
  - updatenotification: 1.11.0
  - user_ldap
  - video_converter: 1.0.5
```

### Nextcloud Signing status

```shell
No errors have been found.
```

**Comment by szaimen** [comment]
From which version did you upgrade?

From 24.0.x to 25.0.3?

**Comment by szaimen** [comment]
Ah, looks like it.

**Comment by OlliZabal** [comment]
Same here after updating from v25.0.2 to v25.0.3
Backend header logo got replaced by the logo.

**Comment by andy-smith-work** [comment]
I had the same logo & image issue after upgrading from v24.0.9 to 25.0.3.  I re-uploaded my images in Settings > Theming.  The **Logo**, **Background & Login Image**, and **Favicon** are good after re-uploading the files, but the **Header Logo** is ignored (the **Logo** is used instead).

**Comment by thomas-svw** [comment]
> I had the same logo & image issue after upgrading from v24.0.9 to 25.0.3. I re-uploaded my images in Settings > Theming. The **Logo**, **Background & Login Image**, and **Favicon** are good after re-uploading the files, but the **Header Logo** is ignored (the **Logo** is used instead).

We have the same problem after upgrading from v24.0.9 to 25.0.3

**Comment by jsalatiel** [comment]
Same here after updating from v25.0.2 to v25.0.3

**Comment by Bouli911** [comment]
Same here after updating from v25.0.2 to v25.0.3

**Comment by DATA-HORIZON** [comment]
This bug report is not about the "Header Logo is ignored". Maybe you should file another issue for that.
Here the problem is the major upgrade from from Nextcloud 24.0.8 to 25.0.2 or 24.0.9 to 25.0.3 that leads to the loss of all theming images.

**Comment by skjnldsv** [comment]
> I had the same logo & image issue after upgrading from v24.0.9 to 25.0.3. I re-uploaded my images in Settings > Theming. The **Logo**, **Background & Login Image**, and **Favicon** are good after re-uploading the files, but the **Header Logo** is ignored (the **Logo** is used instead).

This has been fixed already in upcoming 25.0.4.
Sorry for the troubles :confounded:

**Comment by PVince81** [comment]
Ref: this is the PR that added the scope: https://github.com/nextcloud/server/pull/34599

there were some additional fixes but they were related to migrating user backgrounds from the dashboard app to the user based theme

**Comment by szaimen** [comment]
Will be fixed with 25.0.4 when migrating from 24 to 25. For everyone that already ran into this and didnt revert to 24: you need to manually re-add the assets in the admin theming settings.
